replaced older IM and circumvention charts

GermaVinsmoke opened the pull request:
Signed-off-by: GermaVinsmoke <vaibhav1180@gmail.com>

GermaVinsmoke commented:
Had to add a separate entry point `Chart.js` because the way query and query params are formed are different in the `Chart.js` of `network` page. I think if we can separate out the query generation logic and `useSwr` query hook into separate files then we can generalise the `Chart.js` file into a single `component`.

hellais commented:
This looks great, thanks for putting it together. Some things I think we should do before merging it are:

1. Remove the commented out parts of code and delete them
2. There are several components that are no longer needed and they should also be removed (e.g. `inCountry` context)
3. We need some sort of time interval picker for the charts on the country page (similar to what we have on the network page) so that the user is able to filter by a specified date range
4. Some bits of the page are not translatable (i.e. they need to be wrapped using the `FormattedMessage` component), I will add an inline comment for these
5. Since the country and network page charts are different, to avoid confusion I would give these two components a different name (e.g. `ChartCountry`, `ChartNetwork`) and eventually consider merging them as you said in your comment.

Next steps, which can be done as part of a separate PR:
1. Refactor the coverage chart to make use of the coverage chart we have in the network page (while you are at it I would also address: https://github.com/ooni/explorer/issues/771)
2. As you mentioned in your above comments, look into the possibility of sharing the Chart component between the country and network pages

Other nice to have improvements to the country page include:
* https://github.com/ooni/explorer/issues/424
* https://github.com/ooni/explorer/issues/266

GermaVinsmoke commented:
Hey @hellais , I did 1 and 2. Removed the unused components and files. I wanted to know what to do about some of the section headers, navigation and styling things.

1. The navigation present here, since we removed the Networks section at last, the scroll link doesn't work anymore.
![image](https://user-images.githubusercontent.com/32815509/188273419-f557c64b-be75-41e1-9c9d-66eb435297c7.png)

2. The section headers, are those all good or can we improve that part? Because it still doesn't match - In the Websites section, we got 2 headers of Website.
Then the last 2 charts (Instant Messaging & Circumvention) are present inside Apps section.

hellais commented:
> The navigation present here, since we removed the Networks section at last, the scroll link doesn't work anymore.

Yeah I would say we just drop the networks tab until we re-add support for it.

> The section headers, are those all good or can we improve that part? Because it still doesn't match - In the Websites section, we got 2 headers of Website.

Good catch. I think we would ideally at the very least drop the Websites heading from the Websites section, so we aren't repeating it twice.

> Was thinking do we need to add the filter parameters in URL also?

Yeah I think that's a solid plan

majakomel commented on components/aggregation/mat/timeScaleXAxis.js:
```diff
@@ -5,10 +5,12 @@ import { getDatesBetween } from './computations'
-export function getXAxisTicks (query, count = defaultCount) {
+export function getXAxisTicks (data, count = defaultCount) {
 
-  if (query.axis_x === 'measurement_start_day') {
```
We still need this condition, otherwise the charts with `query.axis_x` other than `measurement_start_day` are broken. With moving query to params the changes here will also not be necessary anymore.

majakomel replied:
I'm referring to your question and Arturo's answer about having parameters in the URL. In that case, this change won't be necessary anymore.

> > Was thinking do we need to add the filter parameters in URL also?
> 
> Yeah I think that's a solid plan

GermaVinsmoke replied:
Hey @majakomel @hellais
Added date and form in the URL of the country page.

GermaVinsmoke replied:
So, now the `ChartCountry` and `ChartNetwork` components are the same, if everything looks good then I can convert those two back to one file.

hellais commented on components/country/ChartCountry.js:
```diff
@@ -0,0 +1,96 @@
+        <Box><Heading h={3} mt={40} mb={20}>{title}</Heading></Box>
+        <Box>
+          {(!chartData && !error) ? (
+            <div> Loading ...</div>
```
This needs to be wrapped with `FormattedMessage`

hellais commented on components/country/ChartCountry.js:
```diff
@@ -0,0 +1,96 @@
+            <div> Loading ...</div>
+          ) : (
+            chartData === null || chartData.length === 0 ? (
+              <Heading h={5}>No Data</Heading>
```
This needs to be wrapped with `FormattedMessage`

hellais commented on components/country/ChartCountry.js:
```diff
@@ -0,0 +1,96 @@
+        {error &&
+          <DetailsBox collapsed={false} content={<>
+            <details>
+              <summary><span>Error: {error.message}</span></summary>
```
This needs to be wrapped with `FormattedMessage`
